google-cloud-clients/google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreOptions.java

```diff
@@ -208,6 +216,32 @@ public Builder setTimestampsInSnapshotsEnabled(boolean value) {
     @Override
     @Nonnull
     public FirestoreOptions build() {
+      if (System.getProperty(FIRESTORE_EMULATOR_SYSTEM_PROPERTY) != null) {
+        String emulatorHost = System.getProperty(FIRESTORE_EMULATOR_SYSTEM_PROPERTY);
+        String hostUrlString = "http://" + emulatorHost;
+        // Try creating a host in order to validate that the host name is valid.
+        try {
+          new URL(hostUrlString);
+          setHost(emulatorHost);
+          setChannelProvider(
+              InstantiatingGrpcChannelProvider.newBuilder()
+                  .setEndpoint(emulatorHost)
+                  .setChannelConfigurator(
+                      new ApiFunction<ManagedChannelBuilder, ManagedChannelBuilder>() {
+                        @Override
+                        public ManagedChannelBuilder apply(ManagedChannelBuilder input) {
+                          input.usePlaintext();
+                          return input;
+                        }
+                      })
+                  .build());
+          setCredentialsProvider(FixedCredentialsProvider.create(new FakeCredentials()));
+        } catch (MalformedURLException e) {
+          throw new IllegalArgumentException(
+              "Value for property FIRESTORE_EMULATOR_HOST is not a valid host", e);
```

Comment:
Can we make this error message more informative?

> The environment variable FIRESTORE_EMULATOR_HOST is unable to be resolved.

Lets also use the constant here instead of duplicating the value.